api/http/api_deployments.go

```diff
@@ -384,6 +430,70 @@ func (d *DeploymentsApiHandlers) getFormFieldValue(p *multipart.Part, maxMetaSiz
 	return &strValue, nil
 }
 
+// ParseGenerateImageMultipart parses multipart/form-data message.
+func (d *DeploymentsApiHandlers) ParseGenerateImageMultipart(mr *multipart.Reader, maxMetaSize int64) (*model.MultipartGenerateImageMsg, error) {
+	multipartGenerateImageMsg := &model.MultipartGenerateImageMsg{}
+	for {
+		p, err := mr.NextPart()
+		if err != nil {
+			return nil, errors.Wrap(err, "Request does not contain artifact")
+		}
+		switch p.FormName() {
+		case "name":
+			name, err := d.getFormFieldValue(p, maxMetaSize)
+			if err != nil {
+				return nil, err
+			}
+			multipartGenerateImageMsg.Name = *name
+		case "description":
+			desc, err := d.getFormFieldValue(p, maxMetaSize)
+			if err != nil {
+				return nil, err
+			}
+			multipartGenerateImageMsg.Description = *desc
+		case "size":
+			size, err := d.getFormFieldValue(p, maxMetaSize)
+			if err != nil {
+				return nil, err
+			}
+			multipartGenerateImageMsg.Size, err = strconv.ParseInt(*size, 10, 64)
+			if err != nil {
+				return nil, err
+			}
+		case "device_types_compatible":
+			deviceTypesCompatible, err := d.getFormFieldValue(p, maxMetaSize)
+			if err != nil {
+				return nil, err
+			}
+			multipartGenerateImageMsg.DeviceTypesCompatible = strings.Split(*deviceTypesCompatible, ",")
+		case "type":
+			fileType, err := d.getFormFieldValue(p, maxMetaSize)
+			if err != nil {
+				return nil, err
+			}
+			multipartGenerateImageMsg.Type = *fileType
+		case "args":
+			args, err := d.getFormFieldValue(p, maxMetaSize)
+			if err != nil {
+				return nil, err
+			}
+			multipartGenerateImageMsg.Args = *args
```

Comment:
@tranchitella is the 'args' still a json string as described in the epic?

am I correct in thinking that:
- deployments just extracts the args, without caring about what's inside
- it passes everything to 'workflows'
- 'workflows' extracts 'args' as json?

Reply:
@mchalski it's an arbitrary string, my idea was to keep it simple and let the consumer figure out what to do with it (e.g., decode it from JSON).